pkg/providers/util/claim_extractor.go

```diff
@@ -94,11 +95,25 @@ func (c *claimExtractor) loadProfileClaims() (*simplejson.Json, error) {
 		return simplejson.New(), nil
 	}
 
-	claims, err := requests.New(c.profileURL.String()).
+	builder := requests.New(c.profileURL.String()).
 		WithContext(c.ctx).
 		WithHeaders(c.requestHeaders).
-		Do().
-		UnmarshalSimpleJSON()
+		Do()
+
+	// We first check if the result is a JWT token
+	// i.e. `application/jwt; charset=utf-8` mime-type
+	mediaType, _, parseErr := mime.ParseMediaType(builder.Headers().Get("Content-Type"))
+
+	if parseErr == nil && mediaType == "application/jwt" {
+		// Decode and use JWT payload as profile claims 
```

Comment:
Please fix the formatting

Reply:
Mmmm, `gofmt` is applied automatically, and does not change anything to the file when applied (?)..